tpm2/include/tpm2_kmyth_io.h

```diff
@@ -72,6 +72,137 @@ int verifyOutputFilePath(char *path);
+ *                                     authorization policy applied to the
+ *                                     sealed data object. This data will be
+ *                                     written to the 'PCR SELECTION LIST'
+ *                                     block in the .ski file after it is
+ *                                     marshalled and base-64 encoded.
+ *
+ * @param[in]  storage_key_public      Pointer to a TPM 2.0 sized buffer
+ *                                     (TPM2B_PUBLIC) holding the data to be
+ *                                     written to the 'STORAGE KEY PUBLIC'
+ *                                     block in the .ski file after it is
+ *                                     marshalled and base-64 encoded.
+ *
+ * @param[in]  storage_key_private     Pointer to a TPM 2.0 sized buffer
+ *                                     (TPM2B_PRIVATE) holding the data to be
+ *                                     written to the 'STORAGE KEY ENC PRIVATE'
+ *                                     block in the .ski file after it is
+ *                                     marshalled and base-64 encoded.
+ *
+ * @param[in]  cipher_string           String specifying the symmetric
+ *                                     encryption method used to encrypt
+ *                                     (wrap) the input data. This string is
+ *                                     written to the 'CIPHER SUITE' block
+ *                                     of the .ski file.
+ *
+ * @param[in]  wrap_key_public         Pointer to a TPM 2.0 sized buffer
+ *                                     (TPM2B_PUBLIC) holding the data to be
+ *                                     written to the 'SYM KEY PUBLIC'
+ *                                     block in the .ski file after it is
+ *                                     marshalled and base-64 encoded.
+ *
+ * @param[in]  wrap_key_private        Pointer to a TPM 2.0 sized buffer
+ *                                     (TPM2B_PRIVATE) holding the data to be
+ *                                     written to the 'SYM KEY ENC PRIVATE'
+ *                                     block in the .ski file after it is
+ *                                     marshalled and base-64 encoded.
+ *
+ * @param[in]  encrypted_data          Symmetrically encrypted data - passed as
+ *                                     a pointer to the array containing the
+ *                                     encrypted data bytes to be written to
+ *                                     the 'ENC DATA' block of the .ski file.
+ *
+ * @param[in]  encrypted_data_size     Size, in bytes, of the encrypted data
+ *                                     to be written to the .ski file.
+ *
+ * @return 0 if success, 1 if error
+ */
+int tpm2_kmyth_create_ski_string(uint8_t ** output,
+                                 size_t *output_length,
+                                 char *filename,
```

Comment:
Why are we tracking filename length here?

Reply:
We don't necessarily know where the filename originates in this function, so we have no guarantee that it ends with a null terminator. We use it in the function in this call:

    concat(&out, &out_length, (uint8_t *) filename, filename_length);